hedera-node/docs/design/services/token-service/airdrops/token-reject.md

```diff
@@ -0,0 +1,135 @@
+# Introduce TokenReject operation
+
+## Purpose
+
+As part of [HIP-904](https://hips.hedera.com/hip/hip-904), we would like to have the option of rejecting a token that has become part of an account's balance (no matter how e.g. via regular crypto transfer or airdrop).
+
+This would mean transferring the token (the whole amount if fungible) or the NFT (concrete serial number from a unique token type) back to its treasury account, without assessing custom fees. Using token reject an account could get rid of potential “ransom” tokens which have very expensive custom fees and only pay for the cheaper token reject operation.
+
+Token reject transaction can be performed only for tokens which are part of the persistent state of the account. No tokens from the pending state that is introduced with HIP-904 could be rejected. This remains only a right for the owner, who can cancel a pending airdrop.
+
+## Goals
+
+1. Define new `TokenReject` HAPI transaction
+2. Implement token reject transaction handler logic
+
+## Non-Goals
+
+- Implement token rejection in system contract functions
+
+## Architecture
+
+The implementation related to the new `TokenReject` transaction will be gated behind a feature flag.
+
+### HAPI updates
+
+Create new transaction type as defined in the HIP:
```

Comment:
Would prefer to link to the protobuf fields in a given tag to ensure any slight changes are in sync.
Can update later or in a future PR when 0.50.0 protobug tag is made

Reply:
Noted, once protobufs is merged and tagged we can add links to the definitions.